btf/btf.go

```diff
@@ -262,19 +267,41 @@ func LoadKernelSpec() (*Spec, error) {
 	return spec.Copy(), nil
 }
 
+// LoadKernelModuleSpec returns the BTF information for the named kernel module.
+//
+// Defaults to /sys/kernel/btf/<module>.
+// Returns an error wrapping ErrNotSupported if BTF is not enabled.
+func LoadKernelModuleSpec(module string) (*Spec, error) {
+	spec, err := kernelModuleSpec(module)
+	if err != nil {
+		return nil, err
+	}
+	return spec.Copy(), nil
+}
+
 var kernelBTF struct {
 	sync.RWMutex
 	spec *Spec
 	// True if the spec was read from an ELF instead of raw BTF in /sys.
 	fallback bool
 }
 
+var kernelModuleBTF struct {
+	sync.RWMutex
+	spec map[string]*Spec
+}
+
 // FlushKernelSpec removes any cached kernel type information.
 func FlushKernelSpec() {
 	kernelBTF.Lock()
+	kernelModuleBTF.Lock()
```

Comment:
Nit: there is another deadlock here if loading a kmod spec means taking kernelBTF.Lock? The locking order there would be kernelModuleBTF, kernelBTF.